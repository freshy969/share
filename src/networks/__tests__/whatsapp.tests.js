import Whatsapp from '../whatsapp';

describe('Share > Whatsapp', () => {
  test('url', () => {
    const whatsapp = Whatsapp.create({});
    whatsapp.entityLink = jest.fn();
    whatsapp.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      whatsapp.url({
        type: 'post',
        id: 60,
        text: 'The Beauties of Gullfoss',
      }),
    ).toMatchSnapshot();

    expect(whatsapp.entityLink.mock.calls).toMatchSnapshot();
    expect(whatsapp.entityLink.mock.results).toMatchSnapshot();
  });
});
