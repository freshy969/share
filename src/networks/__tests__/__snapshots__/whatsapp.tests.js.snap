// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Whatsapp url 1`] = `"https://api.whatsapp.com/send?text=The%20Beauties%20of%20Gullfoss%20https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss"`;

exports[`Share > Whatsapp url 2`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > Whatsapp url 3`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauties-of-gullfoss",
  },
]
`;
